pytorch_lightning/utilities/cli_registries.py

```diff
@@ -0,0 +1,79 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import inspect
+from collections import UserDict
+from typing import Callable, List, Optional, Type
+
+import torch
+
+import pytorch_lightning as pl
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+
+
+class Registry(UserDict):
```

Comment:
This could be integrated with the existing `class _TrainingTypePluginsRegistry(UserDict)` so we avoid duplication

Reply:
Their API and requirement are slightly different. Not sure it makes sense.